### ISSUE TITLE
Some Godog step definitions are never executed, since preceding more general steps will match first

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#922]
So I was facing an issue that sometimes integration tests were failing for me. Today I realized this is because of oc binary download which fails during startup due to exceeded API limit on Github. Issue got resolved for me with setting up MINISHIFT_GITHUB_API_TOKEN...

But even though `minishift start` failed, cucumber step `When executing "minishift start" succeeds` passed. Right now I am not sure if this is problem of only this one step or if it is problem for whole implementation of `executing "xyz" succeeds`.

Log:
```
[agajdosi@dhcp-24-156 minishift]$ make integration
go install -pkgdir=/home/agajdosi/golang/src/github.com/minishift/minishift/out/bindata -ldflags="-X github.com/minishift/minishift/pkg/version.version=1.0.1 -X github.com/minishift/minishift/pkg/version.isoVer
sion=v1.0.2 -X github.com/minishift/minishift/pkg/version.openshiftVersion=v1.5.1" ./cmd/minishift
go test -timeout 3600s github.com/minishift/minishift/test/integration --tags=integration -v -args --binary /home/agajdosi/golang/bin/minishift ""
Running Integration test in: /tmp/minishift-integration-test-715862899
using binary: /home/agajdosi/golang/bin/minishift
....F-----................................F....F......F........F..F..F 70
..F--..F..F..F........                                                 92


--- Failed steps:

    Then  status code of HTTP request to "OpenShift" at "/healthz" is equal to "200" # features/basic.feature:16
      Error: Server returned error: Get https://192.168.42.192:8443/healthz: dial tcp 192.168.42.192:8443: getsockopt: connection refused, on url: https://192.168.42.192:8443/healthz

    Then  stdout should contain # features/basic.feature:57
      Error: Output did not match. Expected: sudoer, Actual:

    Then  stdout should contain # features/basic.feature:70
      Error: Output did not match. Expected: minishift, Actual:

    [...] [...] [...]

22 scenarios (11 passed, 11 failed)
92 steps (74 passed, 11 failed, 7 skipped)
57.30857841s
testing: warning: no tests to run
PASS
exit status 1
FAIL    github.com/minishift/minishift/test/integration 57.316s
Makefile:143: recipe for target 'integration' failed
make: *** [integration] Error 1
```
`/tmp/minishift-integration-test-715862899/cache/oc` is then empty.

**Comment by praveenkumar** [comment]
@agajdosi minishift start passed because vm is up and running so I think we have to check return code for minishift binary when it fails.

**Comment by agajdosi** [comment]
@praveenkumar Yes, I agree it should pass through 3rd step: `Then Minishift should have state "Running"`. But there were changes lately in executing steps: it now has succeeds in the end which I believe is checking for return code of last command.

Current look of scenario:
```
Scenario: Starting Minishift
   Given Minishift has state "Does Not Exist"
   When executing "minishift start" succeeds
    Then Minishift should have state "Running"
```

**Comment by agajdosi** [comment]
Exit code is non-zero when rate limit exceeds:
```
[agajdosi@dhcp-24-156 ~]$ minishift start
Starting local OpenShift cluster using 'kvm' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.2/minishift-b2d.iso'
 40.00 MiB / 40.00 MiB [===========================================================================================================================================] 100.00% 0s
Error starting the cluster:  Error attempting to download and cache oc: Cannot get the OpenShift release version v1.5.0: GET https://api.github.com/repos/openshift/origin/releases/tags/v1.5.0: 403 API rate limit exceeded for 213.175.37.10. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.); rate reset in 48m19.220160315s

[agajdosi@dhcp-24-156 ~]$ echo $?
1
```

So I looked on implementation and the problem is in func executingMinishiftCommandSucceedsOrFails():

```
	success := (lastCommandOutput.ExitCode != 0 || len(lastCommandOutput.StdErr) != 0)
	switch expectedResult {
	case "succeeds":
		if success {
			return fmt.Errorf("Command did not execute successfully. cmdExit: %d, cmdErr: %s", lastCommandOutput.ExitCode, lastCommandOutput.StdErr)
		}
	case "fails":
		if success {
			return fmt.Errorf("Command executed successfully, however was expected to fail. cmdExit: %d, cmdErr: %s", lastCommandOutput.ExitCode, lastCommandOutput.StdErr)
}
```

**Comment by hferentschik** [comment]
Merged via pull request #931
